monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala

```diff
@@ -125,6 +128,18 @@ trait Consumer[-In, +R] extends ((Observable[In]) => Task[R])
     new MapAsyncConsumer[In,R,R2](self, f)
 }
 
+trait PartitionedConsumer[A] {
```

Comment:
In Monix I've had this organization philosophy going against a flat namespace, to keep the root with the essentials for beginners to get started; not necessarily a good idea, but it's also in tune with what RxJava (and other projects that I like) have been doing.

If you think about `PartitionedConsumer`, thing is people are going to discover this type only by bumping into the parameter of `partitionParallel`. So when introducing this particular type, we have to think about where to place it and we could have 3 choices:

1. in the companion object of `Consumer`: in Monix plenty of types are prefixed by companion objects, especially ADTs
2. we could also start a new namespace called `monix.reactive.consumers` and move consumer implementations there - there's already precedent (e.g. `monix.reactive.observers` and `monix.reactive.observables`)
3. this is basically a `Function2` so we shouldn't introduce a new type

Thing is, it's nice to have `PartitionedConsumer.ignore`, however it's not going to be used in practice, because you want to partition stuff in order to do some sort of I/O where that partitioning is useful. Also, this type is prefixed with `Consumer` however it does not inherit from `Consumer`. So personally I would get rid of this type and just work with `Function2`.